lib/block-supports/typography.php

```diff
@@ -72,8 +72,10 @@ function gutenberg_apply_typography_support( $block_type, $block_attributes ) {
 	$has_text_decoration_support = _wp_array_get( $block_type->supports, array( '__experimentalTextDecoration' ), false );
 	$has_text_transform_support  = _wp_array_get( $block_type->supports, array( '__experimentalTextTransform' ), false );
 
+	$skip_font_size_support_serialization = _wp_array_get( $block_type->supports, array( '__experimentalSkipTypographySerialization' ), false );
```

Comment:
The `__experimentalSkipTypographySerialization` flag here doesn't match the `__experimentalSkipFontSizeSerialization` one used in the client-side checks. 

```suggestion
	$skip_font_size_support_serialization = _wp_array_get( $block_type->supports, array( '__experimentalSkipFontSizeSerialization' ), false );
```

Reply:
Ah, good spot, thanks! Silly oversight on my part, since I started this PR by copying some stuff from #30880 😅